File: system-test/dotnet/Dsl.Core/UseCase/External/Tax/TaxDsl.cs
using Driver.Port.External.Tax;
using Dsl.Core.External.Tax.UseCases;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Tax;

public class TaxDsl : IDisposable
{
    private readonly ITaxDriver _driver;
    private readonly UseCaseContext _context;
    private bool _disposed;

    public TaxDsl(ITaxDriver driver, UseCaseContext context)
    {
        _driver = driver;
        _context = context;
    }

    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool disposing)
    {
        if (_disposed) return;
        if (disposing)
            _driver?.Dispose();
        _disposed = true;
    }

    public GoToTax GoToTax() => new(_driver, _context);
    public ReturnsTaxRate ReturnsTaxRate() => new(_driver, _context);
    public GetTaxRate GetTaxRate() => new(_driver, _context);
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/core/usecase/external/tax/TaxDsl.java
package com.mycompany.myshop.testkit.dsl.core.usecase.external.tax;

import com.mycompany.myshop.testkit.common.Closer;
import com.mycompany.myshop.testkit.dsl.core.shared.UseCaseContext;
import com.mycompany.myshop.testkit.dsl.core.usecase.external.tax.usecases.GetTaxRate;
import com.mycompany.myshop.testkit.dsl.core.usecase.external.tax.usecases.GoToTax;
import com.mycompany.myshop.testkit.dsl.core.usecase.external.tax.usecases.ReturnsTaxRate;
import com.mycompany.myshop.testkit.driver.port.external.tax.TaxDriver;

public class TaxDsl implements AutoCloseable {
    private final TaxDriver driver;
    private final UseCaseContext context;

    public TaxDsl(TaxDriver driver, UseCaseContext context) {
        this.driver = driver;
        this.context = context;
    }

    @Override
    public void close() {
        Closer.close(driver);
    }

    public GoToTax goToTax() {
        return new GoToTax(driver, context);
    }

    public ReturnsTaxRate returnsTaxRate() {
        return new ReturnsTaxRate(driver, context);
    }

    public GetTaxRate getTaxRate() {
        return new GetTaxRate(driver, context);
    }
}
